### PR TITLE
Add the ability for caches to use a local site redirector

### PR DESCRIPTION
**Pull request by brianhlin**
A feature for @Mansalu since his CMS XCache container doesn't yet have a site redirector so there's some noise in his logs

### CODE REVIEW

File: atlas-xcache/README.md [hunk 1]
```diff
@@ -24,6 +24,14 @@ XCache:
 - `XC_BLOCKSIZE`: The size of the blocks in the cache
 - `XC_PREFETCH`: Number of blocks to prefetch from a file at once
 
+### Use a local redirector (optional) ###
+
+To configure your cache to make use of a local site redirector, set the following environment variables in the same file
```
Comment (efajardo) [inline atlas-xcache/README.md:29]:
```suggestion
To configure your cache in clustered mode to make use of a cache redirector, set the following environment variables in the same file
```

File: atlas-xcache/README.md [hunk 2]
```diff
@@ -24,6 +24,14 @@ XCache:
...
+as above:
+
+- `XC_REDIRECTOR_FQDN`: the hostname of your local site redirector
```
Comment (efajardo) [inline atlas-xcache/README.md:32]:
```suggestion
- `XC_REDIRECTOR_FQDN`: the hostname of your local cache redirector.
```

File: xcache/xrootd/10-docker-env-var.cfg [hunk 3]
```diff
@@ -5,3 +5,15 @@ set resourcename=$XC_RESOURCENAME
...
+if named stash-origin
+# Stash Origins have this set to redirector.osgstorage.org
+# from the stash-origin RPM packaging
+else if named stash-origin-auth
+# Stash Origins have this set to redirector.osgstorage.org
+# from the stash-origin RPM packaging
+else if defined ?~XC_REDIRECTOR_FQDN && defined ?~XC_REDIRECTOR_PORT
+set REDIRECTOR_FQDN=$XC_REDIRECTOR_FQDN
+set REDIRECTOR_PORT=$XC_REDIRECTOR_PORT
+all.manager $REDIRECTOR_FQDN+ $REDIRECTOR_PORT
```
Comment (efajardo) [inline xcache/xrootd/10-docker-env-var.cfg:18]:
```suggestion
set REDIRECTOR_FQDN1=$XC_REDIRECTOR_FQDN
set REDIRECTOR_PORT1=$XC_REDIRECTOR_PORT
all.manager $REDIRECTOR_FQDN1 $REDIRECTOR_PORT1
all.manager $REDIRECTOR_FQDN2 $REDIRECTOR_PORT2

```

Comment (efajardo) [reply xcache/xrootd/10-docker-env-var.cfg:18]: This sets up for the use of high availability redirectors. Like we do in SoCAL

**Review by efajardo — CHANGES_REQUESTED** [review]
Some minor changes in language and the ability to have multiple redirectors like we do in SoCAL

**Comment by brianhlin** [comment]
@efajardo would anyone ever run more than 2 HA redirectors?

**Comment by efajardo** [comment]
@brianhlin I don't think so. Maybe if data lakes take over people will wanter over redundancy in four or five years. So I think assume two at most is valid.

**Comment by brianhlin** [comment]
@efajardo I went a slightly different route

**Review by efajardo — APPROVED** [review]
It looks good.

**Comment by Mansalu** [comment]
Hooray

**Pull request merged**